packages/typespec-test/test/NetworkAnalytics.Management/generated/typespec-ts/sdk/test/arm-test/src/api/dataProducts/index.ts

```diff
@@ -127,7 +130,9 @@ export async function _listByResourceGroupDeserialize(
 ): Promise<_DataProductListResult> {
   const expectedStatuses = ["200"];
   if (!expectedStatuses.includes(result.status)) {
-    throw createRestError(result);
+    const error = createRestError(result);
+    error.details = errorResponseDeserializer(result.body);
```

Comment:
how about if the exception is not in default shape?

Reply:
this is a common issue that we will not throw rest error if our ser/deserializer failed.

Reply:
https://github.com/Azure/autorest.typescript/issues/2992